ejmask-extensions/src/main/java/com/ebay/ejmask/extenstion/builder/json/JsonBooleanFieldPatternBuilder.java

```diff
@@ -0,0 +1,108 @@
+package com.ebay.ejmask.extenstion.builder.json;
+/**
+ * Copyright (c) 2023 eBay Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.ebay.ejmask.api.PatternEntity;
+import com.ebay.ejmask.extenstion.builder.AbstractRegexPatternBuilder;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.ebay.ejmask.core.util.CommonUtils.emptyIfNull;
+
+/**
+ * An implementation of IPatternBuilder to support sensitive JSON field, whose value need to be partially masked.
+ * This builder is for masking the field value with a Boolean type.
+ *
+ * @author fsun1
+ */
+public class JsonBooleanFieldPatternBuilder extends AbstractRegexPatternBuilder {
+
+    private static final List<PatternEntity> PATTERN_ENTITY_LIST = Arrays.asList(
+            /**
+             * Boolean field with value to be masked
+             * @link https://regex101.com/r/AEwc99/1
```

Comment:
is it the first time we use `@link` in this way?  
as it is not standard usage of `@link`, I am wondering if it causes job action failure.  

https://github.com/eBay/ejmask/actions/runs/11925333247/job/33242123544?pr=20
> Error:  Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.4.1:jar (attach-javadocs) on project ejmask-api: MavenReportException: Error while generating Javadoc: Project contains Javadoc Warnings -> [Help 1]


### reference
- https://www.baeldung.com/javadoc-see-vs-link#link `{@link}` is an inline tag to refer class, method
- if you need refer to web URL, the standard way is to use `@see`, https://www.baeldung.com/javadoc-see-vs-link#2-see-lta-hrefurlgtlabelltagt

Reply:
I know the usage of `@link` is suggested by eBay VAT, but it is not standard way and won't be rendered as webURL in JavaDoc.   

![image](https://github.com/user-attachments/assets/488724f5-0e84-4c75-96da-581aff36bad1)